Compile error with mac os high sierra / Xcode 10.1 toolchain

I was attempting to integrate the Cloud Storage client 0.4.0 in a bazel project on mac os high sierra, with the Xcode 10.1 toolchain. But this compile error repros when I bazel build just the storage_client target directly in the 0.4.0 release folder. It appears the symbol "DEBUG" is being substituted as "1" in a log statement, producing a symbol "GCP_LS_1" which is not recognized.

My toolchain:

>>> g++ -v
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/c++/4.2.1
Apple LLVM version 10.0.0 (clang-1000.11.45.5)
Target: x86_64-apple-darwin18.2.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin


The build error (from within the google-cloud-cpp-0.4.0 directory):

[google-cloud-cpp-0.4.0] >>> bazel build --verbose_failures //google/cloud/storage:storage_client
...
external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O0 -DDEBUG '-std=c++11' -iquote . -iquote bazel-out/darwin-fastbuild/genfiles -iquote bazel-out/darwin-fastbuild/bin -iquote external/bazel_tools -iquote bazel-out/darwin-fastbuild/genfiles/external/bazel_tools -iquote bazel-out/darwin-fastbuild/bin/external/bazel_tools -iquote external/boringssl -iquote bazel-out/darwin-fastbuild/genfiles/external/boringssl -iquote bazel-out/darwin-fastbuild/bin/external/boringssl -iquote external/com_github_curl_curl -iquote bazel-out/darwin-fastbuild/genfiles/external/com_github_curl_curl -iquote bazel-out/darwin-fastbuild/bin/external/com_github_curl_curl -iquote external/com_github_madler_zlib -iquote bazel-out/darwin-fastbuild/genfiles/external/com_github_madler_zlib -iquote bazel-out/darwin-fastbuild/bin/external/com_github_madler_zlib -iquote external/com_github_google_crc32c -iquote bazel-out/darwin-fastbuild/genfiles/external/com_github_google_crc32c -iquote bazel-out/darwin-fastbuild/bin/external/com_github_google_crc32c -isystem google/cloud/storage -isystem bazel-out/darwin-fastbuild/genfiles/google/cloud/storage -isystem bazel-out/darwin-fastbuild/bin/google/cloud/storage -isystem external/boringssl/src/include -isystem bazel-out/darwin-fastbuild/genfiles/external/boringssl/src/include -isystem bazel-out/darwin-fastbuild/bin/external/boringssl/src/include -isystem external/com_github_curl_curl/include -isystem bazel-out/darwin-fastbuild/genfiles/external/com_github_curl_curl/include -isystem bazel-out/darwin-fastbuild/bin/external/com_github_curl_curl/include -isystem external/com_github_madler_zlib -isystem bazel-out/darwin-fastbuild/genfiles/external/com_github_madler_zlib -isystem bazel-out/darwin-fastbuild/bin/external/com_github_madler_zlib -isystem external/com_github_google_crc32c/include -isystem bazel-out/darwin-fastbuild/genfiles/external/com_github_google_crc32c/include -isystem bazel-out/darwin-fastbuild/bin/external/com_github_google_crc32c/include -MD -MF bazel-out/darwin-fastbuild/bin/google/cloud/storage/_objs/storage_client/google_credentials.d '-frandom-seed=bazel-out/darwin-fastbuild/bin/google/cloud/storage/_objs/storage_client/google_credentials.o' -DCURL_STATICLIB -isysroot __BAZEL_XCODE_SDKROOT__ -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' -c google/cloud/storage/oauth2/google_credentials.cc -o bazel-out/darwin-fastbuild/bin/google/cloud/storage/_objs/storage_client/google_credentials.o)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from google/cloud/storage/oauth2/google_credentials.cc:21:
In file included from ./google/cloud/storage/oauth2/authorized_user_credentials.h:18:
In file included from ./google/cloud/storage/internal/curl_request_builder.h:19:
./google/cloud/storage/internal/curl_download_request.h:115:7: error: no member named 'GCP_LS_1' in 'google::cloud::v0::Severity'
      GCP_LOG(DEBUG) << __func__ << "() predicate is false"
      ^~~~~~~~~~~~~~
./google/cloud/log.h:151:26: note: expanded from macro 'GCP_LOG'
  GOOGLE_CLOUD_CPP_LOG_I(GOOGLE_CLOUD_CPP_PP_CONCAT(GCP_LS_, level), \
  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./google/cloud/log.h:93:42: note: expanded from macro 'GOOGLE_CLOUD_CPP_PP_CONCAT'
#define GOOGLE_CLOUD_CPP_PP_CONCAT(a, b) a##b
                                         ^
<scratch space>:64:1: note: expanded from here
GCP_LS_1
^
./google/cloud/log.h:138:43: note: expanded from macro 'GOOGLE_CLOUD_CPP_LOG_I'
               ::google::cloud::Severity::level)>(                         \
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
In file included from google/cloud/storage/oauth2/google_credentials.cc:21:
In file included from ./google/cloud/storage/oauth2/authorized_user_credentials.h:18:
In file included from ./google/cloud/storage/internal/curl_request_builder.h:22:
./google/cloud/storage/internal/curl_upload_request.h:117:7: error: no member named 'GCP_LS_1' in 'google::cloud::v0::Severity'
      GCP_LOG(DEBUG) << __func__ << "() predicate is false"
      ^~~~~~~~~~~~~~
./google/cloud/log.h:151:26: note: expanded from macro 'GCP_LOG'
  GOOGLE_CLOUD_CPP_LOG_I(GOOGLE_CLOUD_CPP_PP_CONCAT(GCP_LS_, level), \
  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./google/cloud/log.h:93:42: note: expanded from macro 'GOOGLE_CLOUD_CPP_PP_CONCAT'
#define GOOGLE_CLOUD_CPP_PP_CONCAT(a, b) a##b
                                         ^
<scratch space>:72:1: note: expanded from here
GCP_LS_1
^
./google/cloud/log.h:138:43: note: expanded from macro 'GOOGLE_CLOUD_CPP_LOG_I'
               ::google::cloud::Severity::level)>(                         \
               ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
2 errors generated.
Target //google/cloud/storage:storage_client failed to build

--- a/google/cloud/log.h
+++ b/google/cloud/log.h
@@ -142,14 +142,16 @@ inline namespace GOOGLE_CLOUD_CPP_NS {
        GOOGLE_CLOUD_CPP_LOGGER_IDENTIFIER.LogTo(sink))                     \
   GOOGLE_CLOUD_CPP_LOGGER_IDENTIFIER.Stream()
 
+// Note that we do not use GOOGLE_CLOUD_CPP_PP_CONCAT here, we actually want
+// to concatenate the literal `level` with `GCP_LS`. In some platforms the level
+// names are used for macros (e.g. on macOS DEBUG is often a macro with value 1
+// for debug builds). If we use GOOGLE_CLOUD_CPP_PP_CONCAT and `level` is a
+// a macro then we would get `GCP_LS_<macro_value>`.
 /**
  * Log a message with the Google Cloud Platform C++ Libraries logging framework.
- *
- *
  */
-#define GCP_LOG(level)                                               \
-  GOOGLE_CLOUD_CPP_LOG_I(GOOGLE_CLOUD_CPP_PP_CONCAT(GCP_LS_, level), \
-                         ::google::cloud::LogSink::Instance())
+#define GCP_LOG(level) \
+  GOOGLE_CLOUD_CPP_LOG_I(GCP_LS_##level, ::google::cloud::LogSink::Instance())
 
 #ifndef GOOGLE_CLOUD_CPP_LOGGING_MIN_SEVERITY_ENABLED
 #define GOOGLE_CLOUD_CPP_LOGGING_MIN_SEVERITY_ENABLED GCP_LS_DEBUG

